src/services/analytics/gtm.ts

```diff
@@ -48,6 +48,23 @@ export const gtmSetChainId = (chainId: string): void => {
   _chainId = chainId
 }
 
+export const enum AbTest {
+  SAFE_CREATION = 'safe-creation',
+}
+
+let _abTest: AbTest | null = null
+
+const gtmResetAbTest = (): void => {
+  _abTest = null
+}
+
+export const gtmSetAbTest = (abTest: AbTest): (() => void) => {
+  _abTest = abTest
+
+  // Return a cleanup function
+  return gtmResetAbTest
+}
```

Comment:
There can be several independent tests running simultaneously. This code allows setting only one.

Also, we don't need to reset it. Tests are global and will not change for a given user.

They should all be initialized in _app.tsx, e.g.
```
useABTest('safe-creation')
useABTest('green-button')
```